### PR TITLE
Chore(DateTimeUtils): copy/move date-time-utils

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

As agreed, we want to move `date-time-utils` to inside `react-northstar` to avoid unnecessary coupling and sharing deps. 

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by pompomon — APPROVED** [review]

File: packages/fluentui/react-northstar/src/utils/date-time-utilities/dateFormatting/formatYear.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as DateGrid from './formatYear';
```
Comment (miroslavstastny) [inline packages/fluentui/react-northstar/src/utils/date-time-utilities/dateFormatting/formatYear.test.ts:1]: react-northstar unit tests execute tests from `test/.*-test`:
https://github.com/microsoft/fluentui/blob/50c2a17463c06292662904065cbeba1ee05122a1/scripts/jest/index.js#L10
That means that none of the moved tests are actually run.
You need to rename and move the test files.

File: packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { validateDate } from 'src/components/Datepicker/validateDate';
-import { DEFAULT_CALENDAR_STRINGS, IRestrictedDatesOptions } from '@fluentui/date-time-utilities';
+import { DEFAULT_CALENDAR_STRINGS, IRestrictedDatesOptions } from 'src/utils/date-time-utilities';
```
Comment (miroslavstastny) [inline packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx:2]: The `src` alias is considered deprecated:
https://github.com/microsoft/fluentui/blob/50c2a17463c06292662904065cbeba1ee05122a1/packages/fluentui/react-northstar/jest.config.js#L6-L8
Use relative path.

**Review by miroslavstastny — CHANGES_REQUESTED** [review]

File: packages/fluentui/react-northstar/src/utils/date-time-utilities/dateFormatting/formatYear.test.ts (hunk 1 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/utils/date-time-utilities/dateFormatting/formatYear.test.ts]: Done

File: packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx (hunk 2 above)
Comment (assuncaocharles) [reply packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx:2]: Trying to use relative path I get `Cannot find module '../../utils/date-time-utils'.ts(2307)`

**Comment by ecraig12345** [comment]
Would you mind explaining a bit more why we decided to go this direction? It seems like a step backwards.

And if we're sure this is the desired direction, can you also get rid of the northstar-specific stuff from within date-time-utilities? (There are also some simplifications you could make to the parts within northstar now that you don't also need to support v8 stuff.)

**Comment by miroslavstastny** [comment]
> Would you mind explaining a bit more why we decided to go this direction? It seems like a step backwards.
> 
> And if we're sure this is the desired direction, can you also get rid of the northstar-specific stuff from within date-time-utilities? (There are also some simplifications you could make to the parts within northstar now that you don't also need to support v8 stuff.)

We discussed shared dependencies with @xugao and @levithomason - the problem with them is that these couple v7/8 and northstar together.

northstar partner is enforcing dependencies' de-duplication. That means that in order to be able to upgrade northstar in their codebase, they would need to upgrade v7/v8 at the same time, which is not doable.

Consuming both v7/8 and northstar is not anything we want to support long term - we want all partners to eventually migrate to v8/next. With that in mind, we do not want to invest any resources into solving the dependency hell and decided NOT to share dependencies between v7/8 and northstar.

**Review by miroslavstastny — APPROVED** [review]

File: packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx (hunk 2 above)
Comment (ecraig12345) [reply packages/fluentui/react-northstar/test/specs/components/Datepicker/validateDate-test.tsx]: In this case you'd need to use ../../../**src**/utils/date-time-utils

**Pull request merged**